modules/auxiliary/gather/censys_search.rb

```diff
@@ -45,18 +45,24 @@ def search(keyword, search_type)
       payload = {
         'query' => keyword
       }
-
-      @cli = Rex::Proto::Http::Client.new('www.censys.io', 443, {}, true)
+      @cli = Rex::Proto::Http::Client.new('search.censys.io', 443, {}, true)
       @cli.connect
 
+    if @searchtype.include?('ipv4')
       response = @cli.request_cgi(
-        'method' => 'post',
-        'uri' => "/api/v1/search/#{search_type}",
+        'method' => 'GET',
+        'uri' => "/api/v2/hosts/search?q=#{keyword}",
+        'headers' => { 'Authorization' => basic_auth_header(@uid, @secret) },
+      )
+      res = @cli.send_recv(response)
+    elsif @searchtype.include?('certificates')
+      response = @cli.request_cgi(
+        'method' => 'GET',
+        'uri' => "/api/v1/view/certificates?#{keyword}",
```

Comment:
`certificates` query doesn't work for me:
```
msf6 auxiliary(gather/censys_search) > run CENSYS_DORK=rapid7 CENSYS_UID=<redacted> CENSYS_SECRET=<redacted> CENSYS_SEARCHTYPE=certificates

[-] Auxiliary failed: JSON::ParserError 859: unexpected token at '<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
...
[-] Call stack:
[-]   /home/msfuser/.rvm/gems/ruby-3.0.2@metasploit-framework/gems/json-2.6.2/lib/json/common.rb:216:in `parse'
[-]   /home/msfuser/.rvm/gems/ruby-3.0.2@metasploit-framework/gems/json-2.6.2/lib/json/common.rb:216:in `parse'
[-]   /home/msfuser/.rvm/gems/ruby-3.0.2@metasploit-framework/gems/activesupport-6.1.6/lib/active_support/json/decoding.rb:23:in `decode'
[-]   /home/msfuser/metasploit-framework/modules/auxiliary/gather/censys_search.rb:76:in `search'
[-]   /home/msfuser/metasploit-framework/modules/auxiliary/gather/censys_search.rb:153:in `run'
[*] Auxiliary module execution completed
```
It looks like this not how this API should be [called](https://search.censys.io/api#/certificates). Apparently, you first need to retrieve the SHA-256 fingerprint of the requested certificate and pass it to `/api/v1/view/certificates`. This fingerprint can be retrieved with the first `/api/v2/hosts/search?q=` query.

Reply:
Ok, i'll get into that tomorrow.